File: example/java/spring-boot/one_consumer_one_provider/consumer/src/main/java/br/com/zup/pact/consumer/service/ClientService.java
package br.com.zup.pact.consumer.service;

import br.com.zup.pact.consumer.dto.BalanceDTO;
import br.com.zup.pact.consumer.dto.ClientDetailsDTO;
import java.util.List;
import java.util.Optional;

public interface ClientService {
    Optional<ClientDetailsDTO> getClientDetails(Integer clientId);

    Optional<List<ClientDetailsDTO>> getAll();

    Optional<BalanceDTO> getBalance(Integer clientId);
}

File: example/java/kotlin/consumer/src/main/kotlin/br/com/zup/pact/consumer/service/ClientService.kt
package br.com.zup.pact.consumer.service

import br.com.zup.pact.consumer.dto.BalanceDTO
import br.com.zup.pact.consumer.dto.ClientDetailsDTO
import java.util.*

interface ClientService {
    fun getClientDetails(clientId: Int): ClientDetailsDTO?
    fun getAll(): List<ClientDetailsDTO>
    fun getBalance(clientId: Int): BalanceDTO?
}
